.travis.yml

```diff
@@ -10,36 +10,38 @@ env:
 - INSTALL_DEB_DEPENDECIES=false
 - INSTALL_DEB_DEPENDECIES=true DUECREDIT_ENABLE=yes
 before_install:
-- wget http://repo.continuum.io/miniconda/Miniconda${TRAVIS_PYTHON_VERSION:0:1}-latest-Linux-x86_64.sh
-  -O /home/travis/.cache/miniconda.sh
-- bash /home/travis/.cache/miniconda.sh -b -p /home/travis/miniconda
-- export PATH=/home/travis/miniconda/bin:$PATH
-- if $INSTALL_DEB_DEPENDECIES; then sudo rm -rf /dev/shm; fi
-- if $INSTALL_DEB_DEPENDECIES; then sudo ln -s /run/shm /dev/shm; fi
-- bash <(wget -q -O- http://neuro.debian.net/_files/neurodebian-travis.sh)
-- sudo apt-get -y update
-- sudo apt-get -y install xvfb fusefat
-- if $INSTALL_DEB_DEPENDECIES; then travis_retry sudo apt-get install -y -qq
-  fsl afni elastix fsl-atlases; fi
-- if $INSTALL_DEB_DEPENDECIES; then
-  source /etc/fsl/fsl.sh;
-  source /etc/afni/afni.sh; fi
-- export FSLOUTPUTTYPE=NIFTI_GZ
+- function bef_inst {
+  wget http://repo.continuum.io/miniconda/Miniconda${TRAVIS_PYTHON_VERSION:0:1}-latest-Linux-x86_64.sh
+  -O /home/travis/.cache/miniconda.sh;
+  bash /home/travis/.cache/miniconda.sh -b -p /home/travis/miniconda;
+  export PATH=/home/travis/miniconda/bin:$PATH ;
+  if $INSTALL_DEB_DEPENDECIES; then sudo rm -rf /dev/shm; fi;
+  if $INSTALL_DEB_DEPENDECIES; then sudo ln -s /run/shm /dev/shm; fi;
+  bash <(wget -q -O- http://neuro.debian.net/_files/neurodebian-travis.sh);
+  sudo apt-get -y update;
+  sudo apt-get -y install xvfb fusefat;
+  if $INSTALL_DEB_DEPENDECIES; then travis_retry sudo apt-get install -y -qq
+  fsl afni elastix fsl-atlases; fi;
+  if $INSTALL_DEB_DEPENDECIES; then
+    source /etc/fsl/fsl.sh;
+    source /etc/afni/afni.sh; fi;
+  export FSLOUTPUTTYPE=NIFTI_GZ; }
```

Comment:
How does failure of one line in this function propagates to the exit code of the whole thing? Since the individual commands are not joined by `&&` wouldn't only the exit code of the last command count in the decision if a retry is needed?